cmake/depends.cmake

```diff
@@ -24,7 +24,13 @@ if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
   find_package(LibRt REQUIRED)
   collect (PROJECT_LIB_DEPS "${LIBRT_LIBRARIES}")
 
-else ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+elseif ("${CMAKE_SYSTEM_NAME}" STREQUAL "NuttX")
+
+  # there is no need to use cmake include detection
+  # under NuttX platform
+  set(HAVE_STDATOMIC_H true)
```

Comment:
Seems not a good idea to force `HAVE_STDATOMIC_H` here
what about ujsrt having 
```
elseif (NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "NuttX")

  # TODO: fix for find_path() to detect stdatomic.h
  # find_path (HAVE_STDATOMIC_H stdatomic.h)
  set (_saved_cmake_required_flags ${CMAKE_REQUIRED_FLAGS})
  set (CMAKE_REQUIRED_FLAGS "-c" CACHE STRING "")
  check_include_files (stdatomic.h HAVE_STDATOMIC_H)
  set (CMAKE_REQUIRED_FLAGS ${_saved_cmake_required_flags})

endif ()
```

Reply:
NuttX platform supports builtin atomic and toolchain atomic.
The reason why CMAKE_REQUIRED_FLAGS cannot be used here is because this cache will destroy the CMake build of the integrated platform

Reply:
My point here is that `HAVE_STDATOMIC_H ` is probably defined in NuttX  so the   `set(HAVE_STDATOMIC_H true)` seems redundant. 
Exclude the cmake code related to `CMAKE_REQUIRED_FLAGS` could be sufficient